src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -78,6 +104,10 @@ double FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::Train(
 {
   ResetData(std::move(predictors), std::move(responses));
 
+  #ifdef NDEBUG
```

Comment:
Should this be `#ifndef`?  Right now this check is only done if `NDEBUG` is defined.